src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -181,21 +193,26 @@ private static IReadOnlyList<TagHelperDescriptor> ResolveTagHelperDescriptors(Ge
             return result;
         }
 
-        private static IReadOnlyList<TagHelperDescriptor> GetTagHelperDescriptorsFromReferences(Compilation compilation, StaticCompilationTagHelperFeature tagHelperFeature)
+        private static IReadOnlyList<TagHelperDescriptor> GetTagHelperDescriptorsFromReferences(GeneratorExecutionContext context, StaticCompilationTagHelperFeature tagHelperFeature)
         {
             List<TagHelperDescriptor> tagHelperDescriptors = new();
+            var compilation = context.Compilation;
 
             foreach (var reference in compilation.References)
             {
                 var guid = reference.GetModuleVersionId(compilation);
-                IReadOnlyList<TagHelperDescriptor> descriptors = default;
+                IReadOnlyList<TagHelperDescriptor> descriptors = new List<TagHelperDescriptor>();
```

Comment:
```suggestion
                var descriptors = new List<TagHelperDescriptor>();
```

Reply:
We need the explicit type here because the out parameter in `TryGetValue` cannot implicitly convert List to `IReadOnlyList`.